Data/NF/Internal.hs

```diff
@@ -51,5 +61,23 @@ newtype NF a
 type role NF nominal
 #endif
 
+-- UGH! GHC swapped the constraint order between
+-- 7.10 and 8.0. If anyone's using some beta compiler,
+-- it might throw up.
+--
+-- Very unfortunately, GHC (as of 8.0) does not allow the pattern
+-- synonym to be *matched* without an NFData context. Hopefully
+-- this will be fixed eventually.
```

Comment:
Hmm, I wonder if this will cause problems. I guess it is not too burdensome to propagate the NFData constraint?